packages/flutter_tools/lib/src/test/test_compiler.dart

```diff
@@ -135,10 +135,10 @@ class TestCompiler {
       );
       // Compilation will fail if there is no flutter_test dependency, since
       // this library is imported by the generated entrypoint script.
-      if (_packageConfig['flutter_test'] == null) {
+      if (_packageConfig['test_api'] == null) {
```

Comment:
The tests have not been updated for this yet, but flutter_test depends on test_api, so this is non-breaking